pkg/rke2/serviceaccount.go

```diff
@@ -0,0 +1,64 @@
+package rke2
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func restrictServiceAccount(ctx context.Context, namespace string, cs *kubernetes.Clientset) error {
+	var sa *v1.ServiceAccount
+	var err error
+	for {
+		sa, err = cs.CoreV1().ServiceAccounts(namespace).Get(ctx, "default", metav1.GetOptions{})
```

Comment:
```suggestion
		sa, err = cs.CoreV1().ServiceAccounts(namespace).Get(ctx, metav1.NamespaceDefault, metav1.GetOptions{})
```

Reply:
This isn't a namespace, its the name of the service account resource. It just happens to be the "default" one.

Reply:
Upstream uses:
```go
import (
  serviceaccountadmission "k8s.io/kubernetes/plugin/pkg/admission/serviceaccount"
)

sa, err = cs.CoreV1().ServiceAccounts(namespace).Get(ctx, serviceaccountadmission.DefaultServiceAccountName, metav1.GetOptions{})

Reply:
Fixed